### DISCUSSION TITLE
What should the default JDK version of the CI be?

### DISCUSSION THREAD

**Discussion by ckipp01** [scala/scala3#17630] [General Question]
I recently came across the discussion in [here](https://github.com/lampepfl/dotty/pull/17044#issuecomment-1454893881) where it was mentioned that the default version we were using for the community build was still 8 and potentially causing issues. From taking a look at the [dotty-ci](https://github.com/lampepfl/dotty-ci) repo, it hasn't been touched in quite some time. Looking at our current workflow file it looks like we're using an image we created over [two years ago!](https://github.com/lampepfl/dotty/blob/28915c4bd20182486b3e11136cf79af516da23b0/.github/workflows/ci.yaml#L48). This leaves me with a few questions since 8 is no longer LTS.

1. We currently install 8 and 17, but set 8 as the default [here](https://github.com/lampepfl/dotty-ci/blob/6305bfa801c7ec042b810c4225688b8035fd0035/dotty-docker/Dockerfile#L18). Do we still want to do this?
2. Even though in the Dockerfile we set 8 to the default, but in our CI it looks like we also [manually set 16?](https://github.com/lampepfl/dotty/blob/28915c4bd20182486b3e11136cf79af516da23b0/.github/workflows/ci.yaml#L66-L67). how come we do this, the git history didn't provide a lot of insight there.
3. We also seem to manually set [Java 8](https://github.com/lampepfl/dotty/blob/28915c4bd20182486b3e11136cf79af516da23b0/.github/workflows/ci.yaml#L472) even though that should be the default.

Regarding the above is there opposition to:

- Updating [dotty-ci](https://github.com/lampepfl/dotty-ci) to new versions of various things, still including Java 8 and Java 17
- No longer set 8 as the default, but 17.
- Leaving the default 17 for everything, but manually setting 8 for the java8 tests.

If there are no issues with this I can dig in.

**Comment by ckipp01** [comment]
Just created https://github.com/lampepfl/dotty-ci/pull/22 to get the ball rolling.

**Comment by sjrd** [comment]
We officially support JDK 8+. It makes sense to me that the default (presumably the version where the most tests are run) be JDK 8.

**Reply by ckipp01** [reply]
Ok, but while it's set as the default originally in the image, we actually manually then set it to 16 for most of the CI. Seeing that the majority is 16, it makes sense to instead just default to 17 in the image, and then instead of manually setting 16, use the default 17 instead no? Also we do have specific tests for 8, so that would remain the same. I don't think the _default_ should really be 8 any longer especially with it being dropped as many places as it is. Of course, we'd still test against 8 in the 8-specific test though.

**Reply by sjrd** [reply]
It seems odd that we're setting it to 16. Perhaps we should revert that, in fact. Or, if absolutely needed, switch to 17, sure.

My impression is that the Community Build tests may require a higher JDK because some libraries may have dropped support for 8. But the rest of the CI should be 8-first, IMO.

Are there actually any tests that are *specific to 8*? In the large majority of cases, if something works on 8, it would work on later JDKs. Tests that are specific to higher JDKs make more sense: they may be tests that rely on particular new features or APIs of a later JDK.
